components/resolved_ts/src/advance.rs

```diff
@@ -70,32 +59,27 @@ impl<T, E: KvEngine> AdvanceTsWorker<T, E> {
             pd_client,
             worker,
             timer: SteadyTimer::default(),
-            raft_router,
             store_meta,
             concurrency_manager,
-            advance_ts_interval,
-            hibernate_regions_compatible,
```

Comment:
Why remove this?

Reply:
We need `CheckLeader` to sync `safe_ts` from leader to follower, the stale read feature can't work if `hibernate_regions_compatible` is `false`,  so we assume it is `true` if the feature is turned on by `resolved-ts.enable`